File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    public int[] solution(int n, int m) {

        return new int[]{GCD(n, m), n * m / GCD(n,m)};
    }

    public int GCD(int a, int b) {
        if (a % b == 0) {
            return b;
        }
        return GCD(b, a % b);
    }
}

File: C#/백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.cs
using System;

namespace Programmers;

public class MainClass
{
    static void Main(string[] args)
    {
        string[] sp = Console.ReadLine().Split(' ');
        int f = int.Parse(sp[0]);
        int s = int.Parse(sp[1]);

        int gcd = GetGCD(f, s);
        int l = f * s / gcd;

        Console.WriteLine(gcd);
        Console.WriteLine(l);
    }

    private static int GetGCD(int f, int s)
    {
        if (f % s == 0)
        {
            return s;
        }

        return GetGCD(s, f % s);
    }
}